### DISCUSSION TITLE
Any plan to have something like Akkagent to integrate multi LLM agents (e.g. stuffs like Microsoft/autogen but Akka.net basis)?

### DISCUSSION THREAD

**Discussion by ingted** [akkadotnet/akka.net#7086] [Ideas]
If we have a Akka.net framework based LLM agents collocation, maybe it could attract more .net developers to embrace Akka.net instead of jump to Python parties....

If there is a plan, I would like to join and contribute some tasks...

**Comment by houstonhaynes** [comment]
This is right up my alley, sort of. 😸 For now I'm using Akka-NET as workers/agents by spawning them whenever they're needed - somewhat of a prescriptive path. 

In an early-days "hello world" scenario that means just using a "parent" actor to hand off the query to a child that's connecting to Ollama through OllamaSharp. That means having a handler that streams the tokens in reponse to a SignalR hub where it gets streamed up to the client. But eventually as the scenarios become more involved that's going to have some form of branching logic - with the idea that we're "keeping receipts" so that a user will know the exact path(s) that a solution took to bring back the most relevant answer. 

In the furthest extreme - that means "breaking out of the box" and doing lots of full-retail calls into individual (presumably better targeted and more efficient) models and/or calculations and just straight up queries where needed - which in my case I'm using Postgresml.org 's current OSS offering as a crutch. Timescale DB is part of the picture too - where I'm putting a vector column with some operational data I'm streaming into have a basis for k-means queries and the like. PostgresML is pretty handy, but also a bit clunky when talking to CPU-only Python models that are "behind the database". 🐌  The GPU accelerated models however are as lively as Ollama. 🦄 Timescale is fantastic and offers as much as PostgresML where the data suits time series analysis.

So - current state I'm mainly hand-jamming activities into child/grand-child actors - but my personal opinion is that for domain-specific problem solving that's much more "grounded" and potentially correct rather than putting the question to black box where all of the "training" happened beforehand.

Let me turn the question back on you @ingted  - what is "Akkagent" and does it already have some model-exercise tooling built into it?

**Reply by ingted** [reply]
Oops... there is no real repo for Akkagent... just for a similarity to Akk-a-gent concept.
